Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -432,14 +427,14 @@ PhysicalParticleContainer::AddGaussianBeam (
         for (long i = 0; i < npart; ++i) {
 #if (defined WARPX_DIM_3D) || (defined WARPX_DIM_RZ)
             const Real weight = q_tot/(npart*charge);
-            const Real x = distx(mt);
-            const Real y = disty(mt);
-            const Real z = distz(mt);
+            const Real x = amrex::RandomNormal(x_m, x_rms, amrex::RandomEngine{});
```

Comment:
@WeiqunZhang should we construct the engine here 3 times or construct it once and pass the object in each of those calls?

Reply:
It's `struct RandomEngine {}; // CPU `.  Moreover I believe the compiler will optimize it out because it's not being used at all.  So the cost is not an issue.  But I would go with `amrex::RandomNormal(x_m, x_rms);`.